### PR TITLE
PLATFORM-8042 update Twiggy extension [Do not remove branch after merging]

### PR DESCRIPTION
**Pull request by mmeller-wikia**
https://fandom.atlassian.net/browse/PLATFORM-8042

### CODE REVIEW

File: src/Hooks.php [hunk 1]
```diff
@@ -1,33 +0,0 @@
...
-	 */
-	public static function onMediaWikiServices(MediaWikiServices $services): void {
-		$services->defineService('TwiggyService', function (MediaWikiServices $services) {
-			$mainConfig = $services->getMainConfig();
-			return new TwiggyService(
-				new FilesystemLoader($mainConfig->get('ExtensionDirectory')),
-				$mainConfig
-			);
-		});
-	}
```
Comment (mmeller-wikia) [inline src/Hooks.php:32]: Moved to ServiceWiring.php

File: composer.json [hunk 2]
```diff
@@ -37,7 +33,8 @@
             "minus-x fix .",
             "phpcbf"
         ],
+        "phpcs": "phpcs",
+        "phan": "vendor/bin/phan -d . --long-progress-bar --load-baseline=.phan-config.php",
```
Comment (mszabo-wikia) [inline composer.json]: That's not a baseline, that's the config file.

File: src/TwiggyService.php [hunk 3]
```diff
@@ -7,51 +7,33 @@
...
-	 * @param string $directory
-	 *
-	 * @return void
-	 */
-	public function setTemplateLocation(string $namespace, string $directory): void {
-		$this->getLoader()->addPath($directory, $namespace);
+		$functionList = array_filter(
+			array_unique( $allowedPhpFunctions ),
+			fn ( $func ): bool => !in_array( $func, $blacklistedPhpFunctions )
+		);
```
Comment (mszabo-wikia) [inline src/TwiggyService.php]: `array_diff( $allowedPhpFunctions, $blacklistedPhpFunctions )`

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: src/PhpFunctionExtension.php [hunk 4]
```diff
@@ -30,53 +31,15 @@
...
 	}
 
-	public function allowFunctions(array $functions)
-	{
-		$this->functions = $functions;
+	/** @inheritDoc */
+	public function getFunctions() {
+		return array_map(
+			static fn ( $function ): TwigFunction => new TwigFunction( $function, $function ),
+			$this->functions
```
Comment (vforge) [inline src/PhpFunctionExtension.php:42]: Is that defined? Shouldn't it be assigned in the constructor?

File: src/TwiggyService.php (hunk 3 above)
Comment (vforge) [reply src/TwiggyService.php]: @mszabo-wikia 👍 is it important that we're losing `array_unique` here?

**Review by vforge — DISMISSED** [review]

File: src/TwiggyService.php (hunk 3 above)
Comment (mszabo-wikia) [reply src/TwiggyService.php]: I think we can keep `array_unique`, I don't expect it to make a big difference but I haven't had time to test yet what Twig would do with duplicate function definitions using the same name.

**Review by mszabo-wikia — APPROVED** [review]

**Comment by maaartyyynaa** [comment]
Please do not delete this branch

**Pull request merged**